compiler/src/dmd/globals.h

```diff
@@ -307,7 +307,7 @@ struct Global
     ErrorSink* errorSink;       // where the error messages go
     ErrorSink* errorSinkNull;   // where the error messages disappear
 
-    FileName (*preprocess)(FileName, const Loc&, bool&, OutBuffer&);
+    DArray<unsigned char> (*preprocess)(FileName, const Loc&, OutBuffer&);
```

Comment:
... whereas on the C++ side DArray [is a trivial type](https://github.com/dlang/dmd/blob/79b2813c53cf26de19d3e37875136a410aebaf7f/compiler/src/dmd/root/dcompat.h#L16) so returns by copy.

Reply:
Changed it to DArray